### PR TITLE
Update navigation dependencies

### PR DESCRIPTION
**Pull request by jtavva**
Update navigation library version to 2.3.0.
Fix for #118 and #35

### CODE REVIEW

File: AndroidTriviaNavigation/app/src/main/java/com/example/android/navigation/TitleFragment.kt [hunk 1]
```diff
@@ -50,13 +50,13 @@ class TitleFragment : Fragment() {
...
 
-    override fun onCreateOptionsMenu(menu: Menu?, inflater: MenuInflater?) {
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
         super.onCreateOptionsMenu(menu, inflater)
-        inflater?.inflate(R.menu.options_menu,menu)
+        inflater.inflate(R.menu.options_menu,menu)
     }
 
-    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
-        return NavigationUI.onNavDestinationSelected(item!!,view!!.findNavController())
```
Comment (kkuan2011) [inline AndroidTriviaNavigation/app/src/main/java/com/example/android/navigation/TitleFragment.kt:59]: code format this file? I think some extra spaces in there could help with readability

File: AndroidTrivia/app/src/main/java/com/example/android/navigation/TitleFragment.kt [hunk 2]
```diff
@@ -50,13 +50,13 @@ class TitleFragment : Fragment() {
...
-    override fun onCreateOptionsMenu(menu: Menu?, inflater: MenuInflater?) {
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
         super.onCreateOptionsMenu(menu, inflater)
-        inflater?.inflate(R.menu.options_menu,menu)
+        inflater.inflate(R.menu.options_menu,menu)
     }
 
-    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
-        return NavigationUI.onNavDestinationSelected(item!!,view!!.findNavController())
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
```
Comment (kkuan2011) [inline AndroidTrivia/app/src/main/java/com/example/android/navigation/TitleFragment.kt:58]: Same code format this file too

File: AndroidTriviaNavigation/app/src/main/java/com/example/android/navigation/TitleFragment.kt (hunk 1 above)
Comment (jtavva) [reply AndroidTriviaNavigation/app/src/main/java/com/example/android/navigation/TitleFragment.kt:59]: this is deleted code

File: AndroidTrivia/app/src/main/java/com/example/android/navigation/TitleFragment.kt (hunk 2 above)
Comment (jtavva) [reply AndroidTrivia/app/src/main/java/com/example/android/navigation/TitleFragment.kt:58]: this is deleted code

**Pull request merged**